gateway/gateway_test.go

```diff
@@ -233,3 +234,33 @@ func TestUpdateReadings(t *testing.T) {
 	test.That(t, err, test.ShouldBeNil)
 	test.That(t, readings, test.ShouldResemble, expectedReadings)
 }
+
+func TestStartCLogging(t *testing.T) {
+	// Create a gateway instance for testing
+	cfg := resource.Config{
+		Name: "test-gateway",
+	}
+
+	// Test case 1: Ensure logging is started if there is no entry in the loggingRoutineStarted map.
+	loggingRoutineStarted = make(map[string]bool)
+
+	g := &gateway{
+		Named:  cfg.ResourceName().AsNamed(),
+		logger: logging.NewTestLogger(t),
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	g.startCLogging(ctx)
+	test.That(t, loggingRoutineStarted["test-gateway"], test.ShouldBeTrue)
+	test.That(t, g.workers, test.ShouldNotBeNil)
+
+	// Test case 2: Ensure no new goroutine is started if the loggingRoutineStarted entry is true.
+	// reset g.workers to test for new workers
+	g.workers = nil
+	g.startCLogging(ctx)
+	test.That(t, g.workers, test.ShouldBeNil)
+	test.That(t, loggingRoutineStarted["test-gateway"], test.ShouldBeTrue)
+
```

Comment:
Let's add a test to ensure that the closure of a sensor removes the logger name from the clogging map.